Confus/Confus/Flag.cpp

```diff
@@ -0,0 +1,171 @@
+#include <irrlicht/irrlicht.h>
+#include <IrrAssimp/IrrAssimp.h>
+
+#include "Flag.h"
+#include "Player.h"
+#include "Collider.h"
+
+
+namespace Confus {
+
+	Flag::Flag(irr::IrrlichtDevice* a_Device, ETeamIdentifier a_TeamIdentifier) : m_TeamIdentifier(a_TeamIdentifier) {
+        //Get drivers to load model
+        auto sceneManager = a_Device->getSceneManager();
+        auto videoDriver = a_Device->getVideoDriver();
+
+        //Load model
+        IrrAssimp irrAssimp(sceneManager);
+        irr::scene::IAnimatedMesh* mesh = sceneManager->getMesh("Media/Meshes/Flag.3ds");
+        m_FlagNode = sceneManager->addAnimatedMeshSceneNode(mesh);
+        m_FlagNode->setMaterialFlag(irr::video::E_MATERIAL_FLAG::EMF_LIGHTING, false);
+        m_FlagNode->setScale({ 2, 2, 2 });
+
+        //Set Color
+		setColor(videoDriver);
+
+        //Add colission box and particle system
+        auto collisionBox = sceneManager->addCubeSceneNode(1.0f, m_FlagNode, 2, irr::core::vector3d<float>(1.f, 2.0f, 0.0f), irr::core::vector3d<float>(0.0f, 0.0f, 0.0f), irr::core::vector3d<float>(2.0f, 4.0f, 1.0f));
+        collisionBox->setVisible(false);
+        initParticleSystem(sceneManager);
+
+		//Add collision callback
+		auto anim = sceneManager->createCollisionResponseAnimator(sceneManager->getActiveCamera()->getTriangleSelector(), collisionBox);
+		Collider col(anim);
+		col.setCallback([this](irr::scene::ISceneNode* a_CollidedNode) 
+		{
+			if ((a_CollidedNode->getID() & 1) == 1) 
+			{
+				captureFlag(ETeamIdentifier::TEAM_RED, NULL);
+				return true;
+			}
+			else if ((a_CollidedNode->getID() & 2) == 2) 
+			{
+				captureFlag(ETeamIdentifier::TEAM_BLUE, NULL);
+				return true;
+			}
+			return false;
+		});
+	}
+
+
+	//Set color & position based on color of flag
+	void Flag::setColor(irr::video::IVideoDriver* a_VideoDriver) 
+	{
+		switch (m_TeamIdentifier)
+		{
+		case ETeamIdentifier::TEAM_BLUE:
+			m_FlagNode->setMaterialTexture(0, a_VideoDriver->getTexture("Media/Textures/Flag/FLAG_BLUE.png"));
+			m_StartPosition = { -2.5f, 3.5f, -2.f };
+			m_StartRotation = { 0.f, 0.f, 0.f };
+			returnToStartPos();
+			break;
+		case ETeamIdentifier::TEAM_RED:
+			m_FlagNode->setMaterialTexture(0, a_VideoDriver->getTexture("Media/Textures/Flag/FLAG_RED.png"));
+			m_StartPosition = { 3.5f, 3.5f, -72.f };
+			m_StartRotation = { 0.f, 180.f, 0.f };
+			returnToStartPos();
+			break;
+		default:
+			m_StartPosition = { 0, 0, 0 };
+			m_StartRotation = { 0, 0, 0 };
+			break;
+		}
+	}
+
+    void Flag::initParticleSystem(irr::scene::ISceneManager* a_SceneManager) {
+        //Create Particle System
+        irr::scene::IParticleSystemSceneNode* ps = a_SceneManager->addParticleSystemSceneNode(false);
+        auto colDark = getColor();
+        auto colBright = colDark;
+        colBright.setAlpha(0);
+
+        //Emitter
+        irr::scene::IParticleEmitter* em = ps->createBoxEmitter(
```

Comment:
I think the coding guidelines ask you to avoid acronyms as much as possible. (ps en em)